toast/tod/sim_detdata.py

```diff
@@ -24,6 +24,82 @@
 from .. import rng as rng
 
 
+def sim_noise_timestream(realization, stream, rate, samples, oversample, freq, psd):
+    
+    fftlen = 2
+    while fftlen <= (oversample * samples):
+        fftlen *= 2
+    half = int(fftlen / 2)
+    norm = rate * float(half)
+    df = rate / fftlen
+
+    interp_freq = np.linspace(df, df*half, num=half, endpoint=True)
+    loginterp_freq = np.log10(interp_freq)
+
+    # ignore zero frequency
+
+    trimzero = False
+    if freq[0] <= 0.0:
+        trimzero = True
+
+    if trimzero:
+        rawfreq = freq[1:]
+    else:
+        rawfreq = freq
+    lograwfreq = np.log10(rawfreq)
+
+    if trimzero:
+        rawpsd = psd[1:]
+    else:
+        rawpsd = psd
+    lograwpsd = np.log10(rawpsd)
+
+    # Ensure that the input frequency range includes all the frequencies
+    # we need.  Otherwise the extrapolation is not well defined.
+
+    if (rawfreq[0] > interp_freq[0]):
+        raise RuntimeError("input PSD does not go to low enough frequency to allow for interpolation")
+    if (rawfreq[-1] < interp_freq[-1]):
+        raise RuntimeError("input PSD does not go to high enough frequency to allow for interpolation")
+
+    # interpolate
+
+    interp = si.InterpolatedUnivariateSpline(lograwfreq, lograwpsd, k=1, ext=2)
```

Comment:
I guess there is no reason we would ever go to higher order?  Hopefully setting k=1 would fall through to a simpler algorithm, but I agree we should just use scipy.interpolate.interp1d for now.